backend/ethereum/glue/eventsub.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package glue
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/pkg/errors"
+)
+
+type (
+	// EventSub generic event subscription.
+	// Can be used on any Contract with any Event.
+	EventSub struct {
+		closed   chan struct{}
+		contract *bind.BoundContract
+		eFact    EventFactory
+
+		watchLogs, filterLogs chan types.Log
+		watchSub, filterSub   event.Subscription
+	}
+
+	// EventType describes the type of an event.
+	EventType struct {
```

Comment:
It's confusing that the Data is part of the type.

Option 1: Merge EventType into Event.
Option 2: Move Data to Event.

It would be good to add a comment for each of the properties to explain what they stand for.

Reply:
Ok i think will merge them. Should be more understandable that way.